scripts/dts/extract_dts_includes.py

```diff
@@ -849,69 +840,93 @@ def main():
-        extract_reg_prop(chosen['zephyr,ccm'], None,
-                         defs, "CONFIG_CCM", 1024, None)
-
-    name_dict = {
-            "CONFIG_UART_CONSOLE_ON_DEV_NAME": "zephyr,console",
-            "CONFIG_BT_UART_ON_DEV_NAME": "zephyr,bt-uart",
-            "CONFIG_UART_PIPE_ON_DEV_NAME": "zephyr,uart-pipe",
-            "CONFIG_BT_MONITOR_ON_DEV_NAME": "zephyr,bt-mon-uart",
-            "CONFIG_UART_MCUMGR_ON_DEV_NAME": "zephyr,uart-mcumgr",
-            }
-
-    for k, v in name_dict.items():
-        if v in chosen:
-            extract_string_prop(chosen[v], None, "label", k, defs)
-
-    # only compute the load offset if a code partition exists and it is not the
-    # same as the flash base address
-    load_defs = {}
-    if 'zephyr,code-partition' in chosen and \
-       'zephyr,flash' in chosen and \
-       reduced[chosen['zephyr,flash']] is not \
-            reduced[chosen['zephyr,code-partition']]:
-        part_defs = {}
-        extract_reg_prop(chosen['zephyr,code-partition'], None, part_defs,
-                         "PARTITION", 1, 'offset')
-        part_base = lookup_defs(part_defs, chosen['zephyr,code-partition'],
-                                'PARTITION_OFFSET')
-        load_defs['CONFIG_FLASH_LOAD_OFFSET'] = part_base
-        load_defs['CONFIG_FLASH_LOAD_SIZE'] = \
-            lookup_defs(part_defs, chosen['zephyr,code-partition'],
-                        'PARTITION_SIZE')
-    else:
-        load_defs['CONFIG_FLASH_LOAD_OFFSET'] = 0
-        load_defs['CONFIG_FLASH_LOAD_SIZE'] = 0
+        defs['dummy-flash'] = {
+            'CONFIG_FLASH_BASE_ADDRESS': 0,
+            'CONFIG_FLASH_SIZE': 0
+        }
 
     insert_defs(chosen['zephyr,flash'], defs, load_defs, {})
 
-    # generate include file
-    if args.keyvalue:
-        generate_keyvalue_file(defs, args)
-    else:
-        generate_include_file(defs, args)
+    return defs
+
+
+def parse_arguments():
+    rdh = argparse.RawDescriptionHelpFormatter
+    parser = argparse.ArgumentParser(description=__doc__, formatter_class=rdh)
+
+    parser.add_argument("-d", "--dts", nargs=1, required=True, help="DTS file")
+    parser.add_argument("-y", "--yaml", nargs=1, required=True,
+                        help="YAML file")
+    parser.add_argument("-f", "--fixup", nargs='+',
+                        help="Fixup file(s), we allow multiple")
+    parser.add_argument("-i", "--include", nargs=1, required=True,
+                        help="Generate include file for the build system")
+    parser.add_argument("-k", "--keyvalue", nargs=1, required=True,
+                        help="Generate config file for the build system")
+    return parser.parse_args()
+
+
+def main():
+    args = parse_arguments()
+
+    dts = load_and_parse_dts(args.dts[0])
+
+    # build up useful lists
+    get_reduced(dts['/'], '/')
+    get_compatibles(dts['/'], '/')
+    get_phandles(dts['/'], '/', {})
+    get_aliases(dts['/'])
+    get_chosen(dts['/'])
```

Comment:
The useful lists are all views on the same DTS. Maybe they can be encapsulated in a DTS class with access methods. That way this class may become an extra module.

Reply:
Indeed, that could enable moving towards a full class-based tool along with your DTDirective class  design.